### PR TITLE
Various catalog related fixes

### PR DESCRIPTION
**Pull request by yesil**
* Add your specific features or fixes

Resolves: 
[MWPW-147697](https://jira.corp.adobe.com/browse/MWPW-147697)
[MWPW-147762](https://jira.corp.adobe.com/browse/MWPW-147762)
[MWPW-147772](https://jira.corp.adobe.com/browse/MWPW-147772)
[MWPW-147850](https://jira.corp.adobe.com/browse/MWPW-147850)
[MWPW-151163](https://jira.corp.adobe.com/browse/MWPW-151163)

**Test URLs:**
- Before: https://main--cc--adobecom.hlx.live/products/catalog/?martech=off&georouting=off
- After: https://catalog-fixes--cc--adobecom.hlx.live/products/catalog?martech=off&georouting=off

### CODE REVIEW

**Review by aishwaryamathuria — APPROVED** [review]

**Comment by Roycethan** [comment]
@yesil Validated above tickets in Catalog https://catalog-fixes--cc--adobecom.hlx.live/products/catalog?martech=off
and fixes looks fine. Although regression on Kitchen sink and other a.com pages is pending - Tomorrow i'll be in whole day AHome related session/meeting, @afmicka will be validating regressions around thanks

**Comment by aishwaryamathuria** [comment]
@yesil @Roycethan If the PR is verified could you change the label to 'verified' and add 'ready for stage' once it can be merged to stage?

**Comment by yesil** [comment]
@aishwaryamathuria I'm waiting for validation from @afmicka  as well.
I'll add the label later today.

File: creativecloud/deps/merch-sidenav.js [hunk 1]
```diff
@@ -345,9 +379,15 @@ customElements.define(
 
 // src/media.js
 var SPECTRUM_MOBILE_LANDSCAPE = "(max-width: 700px)";
-var TABLET_DOWN = "(max-width: 1200px)";
+var TABLET_DOWN = "(max-width: 1199px)";
 
 // src/sidenav/merch-sidenav.js
+document.addEventListener("sp-opened", () => {
+  document.body.classList.add("merch-modal");
```
Comment (npeltier) [inline creativecloud/deps/merch-sidenav.js:386]: 'merch-modal' as constant above?

File: creativecloud/deps/merch-sidenav.js [hunk 2]
```diff
@@ -345,9 +379,15 @@ customElements.define(
...
+});
+document.addEventListener("sp-closed", () => {
+  document.body.classList.remove("merch-modal");
```
Comment (npeltier) [inline creativecloud/deps/merch-sidenav.js:389]: bis

**Comment by npeltier** [comment]
as discussed offline, we should stop with "uber" PR, and have one per ticket

File: creativecloud/deps/merch-sidenav.js [hunk 3]
```diff
@@ -42,13 +42,14 @@ var headingStyles = css`
...
 // src/deeplink.js
+var EVENT_HASHCHANGE = "hashchange";
 function parseState(hash = window.location.hash) {
   const result = [];
   const keyValuePairs = hash.replace(/^#/, "").split("&");
   for (const pair of keyValuePairs) {
     const [key, value = ""] = pair.split("=");
     if (key) {
-      result.push([key, decodeURIComponent(value)]);
+      result.push([key, decodeURIComponent(value.replace(/\+/g, " "))]);
```
[the remark this answers is not in the document]
Comment (yesil) [reply creativecloud/deps/merch-sidenav.js:52]: spaces are converted to +, and are converted back to spaces.

File: creativecloud/deps/merch-sidenav.js [hunk 4]
```diff
@@ -62,15 +63,29 @@ function pushStateFromComponent(component, value) {
 }
 function pushState(state) {
   const hash = new URLSearchParams(window.location.hash.slice(1));
-  Object.entries(state).forEach(([key, value]) => {
-    if (value) {
-      hash.set(key, value);
+  Object.entries(state).forEach(([key, value2]) => {
```
[the remark this answers is not in the document]
Comment (yesil) [reply creativecloud/deps/merch-sidenav.js:66]: it is transpiled code and generated name 🤷‍♂️

File: creativecloud/deps/merch-sidenav.js [hunk 5]
```diff
@@ -62,15 +63,29 @@ function pushStateFromComponent(component, value) {
...
     }
   });
   hash.sort();
-  window.location.hash = decodeURIComponent(hash.toString());
+  const value = hash.toString();
+  let lastScrollTop = window.scrollY || document.documentElement.scrollTop;
+  window.location.hash = value;
+  window.scrollTo(0, lastScrollTop);
+}
+function deeplink(callback) {
```
[the remark this answers is not in the document]
Comment (npeltier) [reply creativecloud/deps/merch-sidenav.js:79]: sorry again just realised where it came from too :)

**Review by npeltier — APPROVED** [review]

**Comment by yesil** [comment]
@aishwaryamathuria this is ready for stage, thanks.

**Comment by aishwaryamathuria** [comment]
@yesil Can you take a look at the failing PSI checks as we cannot merge till the PSI check is successful

**Review by aishwaryamathuria — APPROVED** [review]

**Comment by yesil** [comment]
@aishwaryamathuria the catalog page has heavy UI components and the lighthouse score is not a regression. Same on main.
I propose to merge it as is.
Closer to go-live, we will perform another performance review and address findings if any.

**Pull request merged**